action/protocol/execution/evm/evm.go

```diff
@@ -243,12 +243,12 @@ func ExecuteContract(
 	if err := stateDB.CommitContracts(); err != nil {
 		return nil, nil, errors.Wrap(err, "failed to commit contracts to underlying db")
 	}
-	stateDB.clear()
 	receipt.AddLogs(stateDB.Logs()...).AddTransactionLogs(depositLog, burnLog)
 	if receipt.Status == uint64(iotextypes.ReceiptStatus_Success) ||
 		featureCtx.AddOutOfGasToTransactionLog && receipt.Status == uint64(iotextypes.ReceiptStatus_ErrCodeStoreOutOfGas) {
 		receipt.AddTransactionLogs(stateDB.TransactionLogs()...)
 	}
+	stateDB.clear()
```

Comment:
The position change of this line may be a hard fork.

Reply:
from my understanding it's not a hard-fork, `clear()` does not clear Logs and TransactionLogs, which are used in line 246~249, now moving `clear()` to after line 249 and as a result, these 2 can be cleared as well

it is not a necessary change, just to make `clear()` a little better, that's why I put it in a 2nd commit.

Reply:
then, it should be fine.